### PR TITLE
array_rand() may throw ValueError since 8.0

### PR DESCRIPTION
**Pull request by takaram**
https://3v4l.org/ZAW1B

### CODE REVIEW

**Review by Girgias — APPROVED** [review]
Thank you!

Could you also add the fact that an empty array is affected by the same warning to ValueError promotion?

**Comment by takaram** [comment]
@Girgias 
I added some description. Please review again!
Thank you.

File: reference/array/functions/array-rand.xml [hunk 1]
```diff
@@ -49,11 +51,17 @@
...
-   than there are in the array will result in an
-   <constant>E_WARNING</constant> level error, and NULL will be returned.
   </para>
  </refsect1>
+
+ <refsect1 role="errors">
+  &reftitle.errors;
+  <para>
+   Throws a <classname>ValueError</classname> if <parameter>array</parameter> is empty or <parameter>num</parameter> is
+   out of range.
```
Comment (Girgias) [inline reference/array/functions/array-rand.xml:61]:
```suggestion
   Throws a <classname>ValueError</classname> if <parameter>array</parameter> is empty,
   or if <parameter>num</parameter> is out of range.
```

File: reference/array/functions/array-rand.xml [hunk 2]
```diff
@@ -66,6 +74,14 @@
...
      <tbody>
+      <row>
+       <entry>8.0.0</entry>
+       <entry>
+        <function>array_rand</function> now throws a <classname>ValueError</classname>
+        if <parameter>array</parameter> is empty or <parameter>num</parameter> is out of range;
+        previously <constant>E_WARNING</constant> was raised, and the function returned &null;.
+       </entry>
+      </row>
       <row>
```
Comment (Girgias) [inline reference/array/functions/array-rand.xml:85]:
```suggestion
     <tbody>
      <row>
       <entry>8.0.0</entry>
       <entry>
        <function>array_rand</function> now throws a <classname>ValueError</classname>
        if <parameter>num</parameter> is out of range; previously an
        <constant>E_WARNING</constant> was raised, and the function returned &null;.
       </entry>
      </row>
      <row>
       <entry>8.0.0</entry>
       <entry>
        <function>array_rand</function> now throws a <classname>ValueError</classname>
        if <parameter>array</parameter> is empty; previously an
        <constant>E_WARNING</constant> was raised, and the function returned &null;.
       </entry>
      </row>
      <row>
```

**Review by Girgias — APPROVED** [review]

**Comment by Girgias** [comment]
Thank you!

**Pull request merged**